### PR TITLE
Add Support for 3D Convolution Layer

### PR DESCRIPTION
**Pull request by Ram81**
@deshraj @utsavgarg This PR adds support for 3D Conv layer + import test.

### CODE REVIEW

File: ide/static/js/content.js [hunk 1]
```diff
@@ -366,15 +366,14 @@ class Content extends React.Component {
   exportPrep(callback) {
     this.dismissAllErrors();
     const error = [];
-    const netObj = JSON.parse(JSON.stringify(this.state.net));
-
-    if (Object.keys(netObj).length == 0) {
+    const net_Obj = JSON.parse(JSON.stringify(this.state.net));
```
Comment (kdexd) [inline ide/static/js/content.js]: Minor nitpick - please choose either `camelCase` or `snake_case` and keep it consistent.

File: tensorflow_app/views/import_graphdef.py [hunk 2]
```diff
@@ -55,21 +55,34 @@ def get_padding(node, layer):
...
-
-    pad_h = ((int(output_shape[1]) - 1) * layer['params']['stride_h'] +
-             layer['params']['kernel_h'] - int(input_shape[1])) / float(2)
-    pad_w = ((int(output_shape[2]) - 1) * layer['params']['stride_w'] +
-             layer['params']['kernel_w'] - int(input_shape[2])) / float(2)
-
-    # check this logic (see caffe-tensorflow/kaffe/shapes.py)
-    if node.type == "Conv2D":
+    if node.type == "Conv3D":
+        pad_d = ((int(output_shape[1]) - 1) * layer['params']['stride_h'] +
```
Comment (utsavgarg) [inline tensorflow_app/views/import_graphdef.py]: I think in the params should have been `stride_d` and `kernel_h`.

File: tensorflow_app/views/import_graphdef.py [hunk 3]
```diff
@@ -181,6 +210,20 @@ def import_graph_def(request):
                     except TypeError:
                         return JsonResponse({'result': 'error', 'error':
                                              'Missing shape info in GraphDef'})
+                elif str(node.type) == 'Conv3D':
+                    layer['params']['stride_h'] = int(
+                        node.get_attr('strides')[2])
+                    layer['params']['stride_w'] = int(
+                        node.get_attr('strides')[3])
+                    layer['params']['stride_d'] = int(
```
Comment (utsavgarg) [inline tensorflow_app/views/import_graphdef.py]: Please change order.

File: tensorflow_app/views/import_graphdef.py (hunk 2 above)
Comment (Ram81) [reply tensorflow_app/views/import_graphdef.py]: Oh sorry my bad

**Pull request merged**